src/LuigisBoxBundle/DTO/ConfigDTO.php

```diff
@@ -36,20 +38,27 @@ class ConfigDTO
      */
     private $searchTimeout;
 
+    /**
+     * @var int
+     */
+    private $searchCacheTtl;
+
     public function __construct(
         string $host,
         string $publicKey,
         string $privateKey,
         float $connectionTimeout,
         float $requestTimeout,
-        float $searchTimeout
+        float $searchTimeout,
+        int $searchCacheTtl
     ) {
         $this->host = $host;
         $this->publicKey = $publicKey;
         $this->privateKey = $privateKey;
         $this->connectionTimeout = $connectionTimeout;
         $this->requestTimeout = $requestTimeout;
         $this->searchTimeout = $searchTimeout;
+        $this->searchCacheTtl = min($searchCacheTtl, self::MAX_SEARCH_CACHE_TTL);
```

Comment:
isn't it better to throw an exception instead of silently change provided value?

Reply:
it would be also good to check for negative number

Reply:
check for negative number - done.

I thought about throwing an exception, but chose not to be so strict.
If I had a Logger here, I would like to log it as a warning, but for now I don't.
In fact, I can remove this limit at all, but it does not appear directly from Luigi's documentation. I wanted to show that this is the maximum that can be achieve.